### PR TITLE
native: Add ability to switch sorting method of channels in list

### PR DESCRIPTION
**Pull request by patosullivan**
Fixes TLON-2067

Adds ability to change sorting preference for the channel list (in either the interstitial screen or the channel switcher sheet), stores the user preference in async storage.

Looks like this:
Simulator Screenshot - iPhone 15 Pro - 2024-06-13 at 12 14 43
Simulator Screenshot - iPhone 15 Pro - 2024-06-13 at 12 15 13

Also fixes the issue with the missing unread counts on channels.

### CODE REVIEW

File: packages/ui/src/components/ChannelSwitcherSheet.tsx [hunk 1]
```diff
@@ -23,8 +27,11 @@ export function ChannelSwitcherSheet({
   channels,
   onSelect,
   contacts,
+  sortBy,
+  setSortBy,
 }: Props) {
   const [hasOpened, setHasOpened] = useState(open);
+  // const [sortBy, setSortBy] = useState<'recency' | 'arranged'>('recency');
```
Comment (dnbrwstr) [inline packages/ui/src/components/ChannelSwitcherSheet.tsx:34]: ?

**Review by dnbrwstr — CHANGES_REQUESTED** [review]
Overall looks good to me -- only objection is that you and @latter-bolden are using p different pattern to do pretty similar things in `channelSortPreferences` and `keyVal` https://github.com/tloncorp/tlon-apps/pull/3542/files#diff-e60050fa0b9ae6b31ef3b2930557bc12d2a2bd4e5f5f576ec94079745790fd9c

Could you two sync up on a shared approach for query-invalidate-update for these sorts of settings?

**Comment by jamesacklin** [comment]
Seeing an odd case of mismatched sorting, but this could be just me:

image

**Comment by patosullivan** [comment]
> Seeing an odd case of mismatched sorting, but this could be just me:
> 
> image

yeah, I'm seeing this too, looking into it.

**Comment by patosullivan** [comment]
> > Seeing an odd case of mismatched sorting, but this could be just me:
> > image
> 
> yeah, I'm seeing this too, looking into it.

fixed

**Review by latter-bolden — APPROVED** [review]
Thanks for updating it to match my keyValue!

**Pull request merged**